include/boost/phoenix/stl/cmath.hpp

```diff
@@ -11,37 +11,23 @@
 #include <boost/phoenix/core/limits.hpp>
 #include <cmath>
 #include <boost/phoenix/function/adapt_callable.hpp>
-#include <boost/type_traits/remove_reference.hpp>
-#include <boost/type_traits/remove_cv.hpp>
+#include <boost/type_traits/declval.hpp>
+#include <boost/phoenix/support/iterate.hpp>
 
 namespace boost {
 
 #if (defined (BOOST_NO_CXX11_DECLTYPE) || \
      defined (BOOST_INTEL_CXX_VERSION) || \
              (BOOST_GCC_VERSION < 40500) )
```

Comment:
I suspect that this condition aims to detect trailing result type availability, i.e. `BOOST_NO_CXX11_TRAILING_RESULT_TYPES`.
However, I don't remove that at this time because the PR is opened for fixing a bug.